### PR TITLE
Add DataVolumeSourceVDDK type

### PR DESCRIPTION
**Pull request by mansam**
**What this PR does / why we need it**:
WIP

This PR adds the `DataVolumeSourceVDDK` type and adds support to the
datavolume and import controllers for it. The datasource itself
will need to be done in a follow-up.
See https://github.com/kubevirt/containerized-data-importer/issues/1298

**Special notes for your reviewer**:

**Release note**:

```release-note
Added DataVolumeSourceVDDK type
```

### CODE REVIEW

**Comment by awels** [comment]
Travis is failing on the linter, run make test-lint and make test-unit to make sure both unit tests and linter pass. go fmt -w <file_name> will normally fix any linter issues

File: pkg/apis/core/v1beta1/types.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ type DataVolumeSource struct {
 	Upload   *DataVolumeSourceUpload   `json:"upload,omitempty"`
 	Blank    *DataVolumeBlankImage     `json:"blank,omitempty"`
 	Imageio  *DataVolumeSourceImageIO  `json:"imageio,omitempty"`
+	VDDK     *DataVolumeSourceVDDK     `json:"vddk,omitempty"`
```
Comment (mhenriks) [inline pkg/apis/core/v1beta1/types.go:66]: Please update the alpha API as well.  We committed to supporting alpha for a little longer.

**Comment by awels** [comment]
I see the actual call to the data source is commented out, but that is coming in a different PR right?

**Comment by mansam** [comment]
@awels Yes, @mrnold is working on the data source PR.

**Comment by awels** [comment]
Okay, let me start the functional test conversation now, so we can think about it a little. This PR by itself is not functional but combined with the other PR, we would have to have some way of testing it. The usual way we do a functional test for a data source is create some kind of source in the cluster, create a DV that connects to that source, and see if it succeeds. I am not sure how that would work with this particular source. Any thoughts on how we could test this?

**Comment by mansam** [comment]
I've been thinking about that over the last week or so. There is a vCenter server simulator that I've tried to get nbdkit to talk to, but I don't think it fully implements the VDDK apis. On the other hand, the nbdkit VDDK plugin uses a fake VDDK library to stub out the API calls for testing. We might be able to use something like that so that nbdkit on the importer pod can go through the motions, but we'd have to be very careful about making sure that it accurately represented the behavior of the real thing.

**Comment by awels** [comment]
Okay cool, I just wanted to make sure that was on the radar. We want to ensure that once this is finished we have some functional testing in place, so that when we are maintaining it we have confidence it will remain working. 

One other thing I thought about that recently came up that is not represented here. The types.go is where we define what the fields are and you have updated those properly. However the operator creates the CRDs based on code. For the datavolumes the code is [here](https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/operator/resources/cluster/datavolume.go#L77) and as you can see that is out of sync with the reality in types.go. In #1307 I am syncing them up again, and I have a follow up PR ready to verify that they stay in sync. That will mean depending on which one goes in first, there will be some changes that will have to happen in this PR or #1307

**Comment by mansam** [comment]
Thanks for the heads up. It makes sense to me for this PR to go in after #1307 and just make the appropriate changes here.

**Comment by mrnold** [comment]
This one can probably be closed, since the commits were included in #1320.

**Comment by awels** [comment]
Closed by request

**Pull request closed without merging**